lib/mongoid/includes/inclusion.rb

```diff
@@ -17,6 +17,14 @@ def nested?
         !!from
       end
 
+      # Public: Checks if the collection already has an inclusion with the
+      # specified metadata.
+      def eql?(other)
+        metadata == other && from == other.from
+      rescue
+        false
```

Comment:
Using `respond_to?` instead of `try` should do the trick. Let's avoid throwing an exception 😃